auth/store.go

```diff
@@ -569,7 +569,12 @@ func (as *authStore) RoleGrantPermission(r *pb.AuthRoleGrantPermissionRequest) (
 	return &pb.AuthRoleGrantPermissionResponse{}, nil
 }
 
-func (as *authStore) isOpPermitted(userName string, key string, write bool, read bool) bool {
+func isSubset(begin1, end1, begin2, end2 []byte) bool {
+	// return true if [begin1, end1) is a subset of [begin2, end2)
+	return 0 <= bytes.Compare(begin1, begin2) && bytes.Compare(begin2, end2) <= 0
```

Comment:
Did you mean `bytes.Compare(end1, end2) <= 0`?

Otherwise, `end1 > end2` is fine?